examples_tests/43.AllocatorTest/main.cpp

```diff
@@ -1,53 +1,219 @@
+	struct EntryForFrameData
+	{
+		uint32_t outAddr[maxAllocs] = { ~0u };
+		uint32_t sizes[maxAllocs] = { 0u };
+		uint32_t alignments[maxAllocs] = { 0u };
+	} perFrameData;
+	
+	void executeAllocatorTest()
+	{
+		os::Printer::log("Executing Pool Allocator Test!", ELL_INFORMATION);
+	
+		for (size_t i = 0; i < creationParameters.multiAllocCnt; ++i)
+			executeForFrame();
+	}
+
+private:
+	// function to test allocator
+		// pick random alignment, rand buffer size (between 2kb and 2GB), random offset (less than buffer size), random alignment offset (less than alignment) and other parameters randomly
+
+		// allocate reserved space (for allocator state)
+
+		// create address allocator
+
+		// randomly decide the number of iterations of allocation and reset
+			// declare `core::vector` to hold allocated addresses and sizes
+
+			// randomly decide how many `multi_allocs` to do
+				// randomly decide how many allocs in a `multi_alloc` NOTE: must pick number less than `traits::max_multi_alloc`
+					// randomly decide sizes (but always less than `address_allocator_traits::max_size`)
+				// record all successful alloc addresses to the `core::vector`
+
+				// run random dealloc function
+			//
+
+			// run random dealloc function
+
+			// randomly choose between reset and freeing all `core::vector` elements
+				// reset
+			// ELSE
+				// free everything with a series of multi_free
+	void executeForFrame()
+	{
+		uint32_t randMaxAlign = rng.getRndMaxAlign();
+		uint32_t randAddressSpaceSize = rng.getRndBuffSize();
+		uint32_t randAlignOffset = rng.getRandomNumber(0u, randMaxAlign - 1u);
+		uint32_t randOffset = rng.getRandomNumber(0u, randAddressSpaceSize - 1u); //?
+		uint32_t randBlockSz = rng.getRandomNumber(0u, randAddressSpaceSize - 1u); //?
+
+		uint32_t allocationAmountForSingeFrameTest = rng.getRndAllocCnt();
+		
+		const auto reservedSize = PoolAddressAllocator::reserved_size(randMaxAlign, randAddressSpaceSize, randBlockSz); // 3rd parameter onward is custom for each address alloc type
+		void* reservedSpace = _IRR_ALIGNED_MALLOC(reservedSize, _IRR_SIMD_ALIGNMENT);
+		auto poolAlctr = PoolAddressAllocator(reservedSpace, randOffset, randAlignOffset, randMaxAlign, randAddressSpaceSize, randBlockSz);
+
+		// randomly decide how many `multi_allocs` to do
+		const uint32_t multiAllocCnt = rng.getRandomNumber(1u, 5u); //range?
+		for (uint32_t i = 0u; i < multiAllocCnt; i++)
+		{
+			// randomly decide how many allocs in a `multi_alloc` NOTE: must pick number less than `traits::max_multi_alloc`
+			const uint32_t allocCntInMultiAlloc = rng.getRandomNumber(1u, 4096u); //?
+
+			for (size_t i = 0u; i < allocCntInMultiAlloc; ++i)
+			{
+				// randomly decide sizes (but always less than `address_allocator_traits::max_size`)
+				perFrameData.sizes[i] = rng.getRandomNumber(1u, Traits::max_size(poolAlctr));
+				perFrameData.alignments[i] = rng.getRndMaxAlign();
```

Comment:
pool allocator cant do random sizes and alignments, they always need to be block sized... other allocators can though.

Reply:
also you need to choose the rand alignment differently than the max for the allocator
https://github.com/Devsh-Graphics-Programming/Nabla/pull/1/files/0078e583bc8f69dd8697b9a6d5eb5e35f0aaa769#r496261998